server/service/src/auth.rs

```diff
@@ -483,9 +486,12 @@ fn all_permissions() -> HashMap<Resource, PermissionDSL> {
     );
     map.insert(
         Resource::MutateProgram,
```

Comment:
Could make a specific one for `MutateImmunisationProgram`?

Reply:
You are not using the `Resource::MutateProgram` resource for GAPS do you? If not you don't need to make changes here.

You already have the `Resource::MutateVaccineCourse` where you can define required permissions...

Side note, you can also use the dynamic permission which works over the `context`, i.e. if your program has a context id the user needs permissions for this context to get access. For example, to get `VaccineCourseMutate` access to a hypothetical program "VaccineCourse2Program" the user needs permissions for the matching  "VaccineCourse2Program" context. Which is defined like:
`PermissionDSL::HasDynamicPermission(PermissionType::VaccineCourseMutate),`

Configured in mSupply like (but UI only support `PermissionType::{DocumentMutate|DocumentQuery}` at the moment):
![image](https://github.com/msupply-foundation/open-msupply/assets/88299195/434fec3c-6d30-4fa3-bf36-5ba8e2001259)

Reply:
Seems like maybe we do want `MutateImmunisationProgram`... can't see why you would be able to mutate immunization programs but not vaccine courses?

Reply:
yes a Resource::MutateVaccineCourse and a PermissionType::ImmunisationProgram makes sense to me (because its a build in special program and not a "dynamic" patient program)

Reply:
Thanks all, I'll look at using context in future if more complicated permissions are required.